auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.firebase.ui.auth.ui.email;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import com.firebase.ui.auth.util.data.AuthViewModelBase;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+public class RecoverPasswordHandler extends AuthViewModelBase {
+    private Pair<String, Task<Void>> mCachedPasswordReset;
+
+    public RecoverPasswordHandler(Application application) {
+        super(application);
+    }
+
+    public void startReset(final String email) {
+        if (mCachedPasswordReset == null || !TextUtils.equals(email, mCachedPasswordReset.first)) {
```

Comment:
Can you explain a little bit what's going on here?

Reply:
The idea was to cache the task so that if the user mashed the "Send" button, we would only start the request once, but I thought about this and meh. It's extra complexity for something that would force the user to be stuck in an error state so I removed it.